docs/source/best-practices/checkpointing.rst

```diff
@@ -25,25 +25,37 @@ Checkpointing is a fault tolerance technique designed to overcome the “fail-st
+  * Model-level checkpointing - training models are often the intent of the user doing machine learning on Discovery.
+
+Which checkpointing method to use?
+----------------------------------
+ * If your software already comes with built-in checkpointing, it is often the preferred option. It is the most optimized and efficient way to the checkpoint.
+ * **Application-level** checkpointing is the easiest to use, as it exists in your application: it does not require significant changes to your scripts.
+ * **Application-level** checkpointing will save only the relevant data for your specific application.
+ * If you're writing your code - use DMTCP or implement your own checkpointing.
+ * **ML Model-level** checkpointing is specific to model training and deployment (see `ML Model-level`_)
 
 .. note::
-  If you are developing code using Python, Matlab or R, there are packages and functions that can be used to implement checkpointing easily. Some examples include `Python PyTorch checkpointing <https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_a_general_checkpoint.html>`_, `TensorFlow checkpointing <https://www.tensorflow.org/guide/checkpoint>`_, `Python Pickle checkpointing <https://deap.readthedocs.io/en/master/tutorials/advanced/checkpoint.html>`_, `MATLAB checkpointing <https://www.mathworks.com/help/gads/work-with-checkpoint-files.html>`_ and `R checkpointing <https://cran.r-project.org/web/packages/checkpoint/vignettes/checkpoint.html>`_. Additionally, many Computational Chemistry and Molecular Dynamics software have built-in checkpointing options, such as `GROMACS <https://manual.gromacs.org/documentation/current/user-guide/managing-simulations.html>`_ and `LAMMPS <https://docs.lammps.org/restart.html>`_.  
+  Some packages and functions allow for easy checkpointing if you are developing code using Python, Matlab, or R. Some examples include `Python PyTorch checkpointing <https://pytorch.org/tutorials/recipes/recipes/saving_and_loading_a_general_checkpoint.html>`_, `TensorFlow checkpointing <https://www.tensorflow.org/guide/checkpoint>`_, `Python Pickle checkpointing <https://deap.readthedocs.io/en/master/tutorials/advanced/checkpoint.html>`_, `MATLAB checkpointing <https://www.mathworks.com/help/gads/work-with-checkpoint-files.html>`_ and `R checkpointing <https://cran.r-project.org/web/packages/checkpoint/vignettes/checkpoint.html>`_. Additionally, many Computational Chemistry and Molecular Dynamics software have built-in checkpointing options, such as `GROMACS <https://manual.gromacs.org/documentation/current/user-guide/managing-simulations.html>`_ and `LAMMPS <https://docs.lammps.org/restart.html>`_.
 
 
-Implementing checkpointing can be acheived by:
- * Some save-and-load mechanism of your calculation state. 
- * The use of `Slurm Job Arrays <https://slurm.schedmd.com/job_array.html>`_. 
+Implementing checkpointing can be achieved by:
```

Comment:
Can we change this to "Implement checkpointing:" or "How to Implement checkpointing?"